openmc/mgxs_library.py

```diff
@@ -0,0 +1,2577 @@
+import copy
+from numbers import Real, Integral
+import os
+
+import numpy as np
+import h5py
+from scipy.interpolate import interp1d
+from scipy.integrate import simps
+from scipy.special import eval_legendre
+
+import openmc
+import openmc.mgxs
+from openmc.checkvalue import check_type, check_value, check_greater_than, \
+    check_iterable_type, check_less_than, check_filetype_version
+
+
+# Supported incoming particle MGXS angular treatment representations
+_REPRESENTATIONS = ['isotropic', 'angle']
```

Comment:
Since the strings `isotropic` and `angle` are later used directly in code, it's probably a good idea to define constants now and use those instead of the raw string for all possible usages in code. 

The main benefit is that a *very* common source of bugs is the result of mispellings, for example, `if representation == 'angel'`. Constants help avoid that risk.

The code would look something like this:

```
REPRESENTATION_ISOTROPIC = 'isotropic'
REPRESENTATION_ANGLE = 'angle'

_REPRESENTATIONS = [
    REPRESENTATION_ISOTROPIC,
    REPRESENTATION_ANGLE,
]
```

It might look overwhelming now, but later on in the code references to `REPRESENTATION_ISOTROPIC` are helpful bc it provides context to the value.

An alternative is to not prepend with `REPRESENTATION`.

```
ISOTROPIC = 'isotropic'
ANGLE = 'angle'

_REPRESENTATIONS = [
    ISOTROPIC,
    ANGLE,
]
```

This provides the benefits of using constants but without the context that the constants refer to the representations.